core/src/main/java/org/wso2/testgrid/core/configchangeset/ConfigChangeSetExecutorUnix.java

```diff
@@ -0,0 +1,242 @@
+     * @param initShellCommand  List of shell command
+     * @return  True if execution success. Else, false
+     */
+    private boolean applyShellCommandOnAgent(TestPlan testPlan, List<String> initShellCommand) {
+        String tinkererHost = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_REST_BASE_PATH);
+        // Get authentication detail from config.properties and encode with BASE64
+        String authenticationString = ConfigurationContext.getProperty(
+                ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_USERNAME) + ":" +
+                ConfigurationContext.getProperty(
+                        ConfigurationContext.ConfigurationProperties.DEPLOYMENT_TINKERER_PASSWORD);
+        String authenticationToken = "Basic " + Base64.getEncoder().encodeToString(
+                authenticationString.getBytes(StandardCharsets.UTF_8));
+        boolean executionStatus = true;
+        String agentLink = tinkererHost + "test-plan" + TestGridConstants.FILE_SEPARATOR + testPlan.getId() +
+                TestGridConstants.FILE_SEPARATOR + "agents";
+        try {
+            // Get list of agent for given test plan id
+            Content agentResponse = Request.Get(agentLink)
+                    .addHeader(HttpHeaders.AUTHORIZATION, authenticationToken).
+                            execute().returnContent();
+
+            Agent[] agents = new Gson().fromJson(agentResponse.asString(), Agent[].class);
+            // Execute command on found agents
+            for (Agent agent : Arrays.asList(agents)) {
+                logger.info("Start sending commands to agent " + agent.getAgentId());
+                for (String shellCommand : initShellCommand) {
+                    logger.info("Execute: " + shellCommand);
+                    Content shellCommandResponse = sendShellCommand(tinkererHost, authenticationToken
+                            , agent, shellCommand);
+                    JsonParser jsonParser = new JsonParser();
+                    JsonObject result = (JsonObject) jsonParser.parse(shellCommandResponse.asString());
+                    // Set default  value as empty
+                    int exitValue = 0;
+                    String code = "";
+                    String response = "";
+                    if (result.get("exitValue") != null) {
+                        exitValue = result.get("exitValue").getAsInt();
+                    }
+                    if (result.get("code") != null) {
+                        code = result.get("code").getAsString();
+                    }
+                    if (result.get("response") != null) {
+                        response = result.get("response").getAsString();
+                    }
+                    logger.info("Agent exit value: " + exitValue);
+                    logger.info("Agent code: " + code);
+                    logger.info("Agent response: \n" + response);
```

Comment:
Above logs are missing the context. When you have a log file with thousands of lines, the context becomes important. In here, we should print what's the testplan and agent, which config set you are trying to apply etc. 

We can merge these into one log line as well imo.